### PR TITLE
fix: check for old tmp table before deleting it

### PR DESCRIPTION
**Pull request by octiva**
# Description

Implemented as discussed in #546.

Uses `adapter.get_relation` to search for an old tmp_relation that could exist if the previous table creation did not clean it up properly. 

## Checklist

- [X] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [X] You kept your Pull Request small and focused on a single feature or bug fix.
- [ ] You added unit testing when necessary
- [ ] You added functional testing when necessary

### CODE REVIEW

**Comment by nicor88** [comment]
Could you double check if we need to adapt also anything in the [incremental part](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/incremental/incremental.sql)? maybe we can benefit for the same approach too, that seems more resilient and correct.

File: dbt/include/athena/macros/materializations/models/incremental/incremental.sql [hunk 1]
```diff
@@ -74,9 +76,8 @@
         {% do exceptions.raise_compiler_error(inc_predicates_not_list) %}
       {% endif %}
     {% endif %}
-    {% set tmp_relation = make_temp_relation(target_relation) %}
```
Comment (nicor88) [inline dbt/include/athena/macros/materializations/models/incremental/incremental.sql:77]: any specific reason why this was moved on top?

**Review by nicor88 — COMMENTED** [review]
Thanks for this improvement @octiva 💯 

@svdimchenko / @Jrmyy could you have another look?

File: dbt/include/athena/macros/materializations/models/incremental/incremental.sql [hunk 2]
```diff
@@ -11,7 +12,10 @@
   {% set force_batch = config.get('force_batch', False) | as_bool -%}
   {% set target_relation = this.incorporate(type='table') %}
   {% set existing_relation = load_relation(this) %}
-  {% set tmp_relation = make_temp_relation(this) %}
+  {% set old_tmp_relation = adapter.get_relation(identifier=identifier ~ '__dbt_tmp',
+                                             schema=schema,
+                                             database=database) %}
+  {% set tmp_relation = make_temp_relation(target_relation) %}
```
Comment (nicor88) [inline dbt/include/athena/macros/materializations/models/incremental/incremental.sql:18]: given the fact that you use `__dbt_tmp` for `old_tmp_relation`, shall we pass explicitely also `__dbt_tmp` as second argument to in `make_temp_relation `? e.g. `make_temp_relation(target_relatiom, '__dbt_tmp')`
just to have a crystal clear "table" suffix for tmp tables, and to align on what we do in the "table materialization", where we set table suffix esplicitely.

File: dbt/include/athena/macros/materializations/models/incremental/incremental.sql (hunk 1 above)
Comment (spencer-horton-wcq) [reply dbt/include/athena/macros/materializations/models/incremental/incremental.sql:77]: It was re-assigned multiple times to `make_temp_relation(target_relation)`, and never used the original assignment, I thought it was a clean and easy refactor

File: dbt/include/athena/macros/materializations/models/incremental/incremental.sql (hunk 2 above)
Comment (spencer-horton-wcq) [reply dbt/include/athena/macros/materializations/models/incremental/incremental.sql:18]: @nicor88 Yeah for sure, then we aren't reliant on the default suffix.

Comment (octiva) [reply dbt/include/athena/macros/materializations/models/incremental/incremental.sql]: @nicor88 I have updated this now :)

Comment (nicor88) [reply dbt/include/athena/macros/materializations/models/incremental/incremental.sql]: > Yeah for sure, then we aren't reliant on the default suffix.
Exactly, doing so the code is more robust to default upstream changes in dbt-core.
Thanks for adjusting the code.

**Review by Jrmyy — APPROVED** [review]

**Review by nicor88 — APPROVED** [review]
Thanks for your contribution @octiva

**Comment by spencer-horton-wcq** [comment]
Happy to contribute!

**Pull request merged**